dodo.py

```diff
@@ -81,20 +81,81 @@ def task_openspiel_compile():
     }
 
 
-def task_forecast():
+#####################
+# FORECASTING TASKS #
+#####################
+
+FORECAST_QUERY_LOG_DIR = Path("./forecast/data/extracted/extended/")
+
+FORECAST_ARTIFACTS_DIR = Path("artifacts")
+FORECAST_PREPROCESSOR_ARTIFACT = FORECAST_ARTIFACTS_DIR.joinpath(
+    "preprocessed.parquet.gzip"
+)
+FORECAST_CLUSTER_ARTIFACT = FORECAST_ARTIFACTS_DIR.joinpath("clustered.parquet")
+FORECAST_MODEL_DIR = FORECAST_ARTIFACTS_DIR.joinpath("models")
+FORECAST_PREDICTION_CSV = FORECAST_ARTIFACTS_DIR.joinpath("forecast.csv")
+
+FORECAST_PREDICTION_START = "2021-12-06 14:24:32 EST"
+FORECAST_PREDICTION_END = "2021-12-06 14:24:36 EST"
+
+
+def task_forecast_preprocess():
     """
-    Forecast: forecast the workload.
+    Forecast Preprocess: create templatized queries data for ingest into
+    forecaster training or prediction
     """
+
+    preprocessor_action = (
+        "python3 ./forecast/preprocessor.py"
+        f" --query-log-folder {FORECAST_QUERY_LOG_DIR.absolute()} "
+        f"--output-parquet {FORECAST_PREPROCESSOR_ARTIFACT.absolute()}"
+    )
+
     return {
         "actions": [
-            "mkdir -p artifacts",
+            f"mkdir -p {FORECAST_ARTIFACTS_DIR.absolute()}",
             # Preprocess the PostgreSQL query logs.
-            "python3 ./forecast/preprocessor.py --query-log-folder ./forecast/data/extracted/extended/ --output-hdf ./artifacts/preprocessor.hdf",
-            # Cluster the processed query logs.
-            # TODO(WAN): clusterer shouldn't go directly to the predictions, plug in Mike's part.
-            "python3 ./forecast/clusterer.py --preprocessor-hdf ./artifacts/preprocessor.hdf --output-csv ./artifacts/forecast.csv",
+            preprocessor_action,
+        ],
+        "targets": [FORECAST_PREPROCESSOR_ARTIFACT.absolute()],
+        "uptodate": [False],  # TODO(WAN): Always recompute?
+        "verbosity": VERBOSITY_DEFAULT,
+    }
+
+
+def task_forecast_predict():
+    """
+    Forecast: forecast the workload.
+    """
+
+    cluster_action = (
+        "python3 ./forecast/clusterer.py"
+        f" --preprocessor-parquet {FORECAST_PREPROCESSOR_ARTIFACT.absolute()}"
+        f" --output-parquet {FORECAST_CLUSTER_ARTIFACT.absolute()}"
+    )
+
+    forecast_action = (
+        "python3 ./forecast/forecaster.py"
+        f" -p {FORECAST_PREPROCESSOR_ARTIFACT.absolute()}"
+        f" -c {FORECAST_CLUSTER_ARTIFACT.absolute()}"
+        f" -m {FORECAST_MODEL_DIR.absolute()}"
+        f' -s "{FORECAST_PREDICTION_START}"'
+        f' -e "{FORECAST_PREDICTION_END}"'
+        f" --output_csv {FORECAST_PREDICTION_CSV.absolute()}"
+        " --override_models"
+    )
+
+    return {
+        "actions": [
+            cluster_action,  # Cluster the processed query logs.
```

Comment:
Nit: I generally prefer comments on their own line since it makes scanning downwards easier, but I'm OK with this if you prefer it this way.